submissions/krylenger/frogger/app.js

```diff
@@ -0,0 +1,246 @@
+const borders = {
+    topY: 49,
+    rightX: 420,
+    bottomY: 464,
+    leftX: 20 
+}
+
+const tileDimensions = {
+    width: 100,
+    height: 83
+}
+
+const score = document.querySelector('.score');
+const lives = document.querySelector('.lives');
+
+const randomSpeed = () => {
+    return Math.floor(Math.random() * (350 - 100 + 1)) + 100;
+}
+
+//////////  Enemy /////////////////
+
+const Enemy = function() {
+    this.width = 70;
+    this.height = 83;
+    this.speed = randomSpeed();
+};
+
+Enemy.prototype.handleGameOver = function () {
+  if (
+    this.x + this.width > player.x &&
```

Comment:
Classes should not refer to any global variables, like global variable `player`, which is an instance of Player class
(referring to global constants and globals provided by the gaming platform like Resources is OK);
Hint: pass Player instance as an argument to every enemy

Reply:
Thank you for your advices so much!
This helped me to understand the key principles of OOP pattern better and to see some weak and non logical parts in my code (such as some methods of Player class were attached to Enemy class).
What about this part: 
`Player.prototype.resetGame = function() {
  alert(`You lose :( Your score is ${this.score} Another try?`);
  this.lives = 3;
  this.score = 0;
  score.textContent = `Score: ${this.score}`;
  lives.textContent = `Lives: ${this.lives}`;
}`
is it fine to update score and lives global variables which are DOM elements from inside the Class?

Reply:
This is a very good question. I think this is out of scope of this particular task (your solution looks fine to me), but in a general case you would probably have a dedicated entity for score board with methods like `setScore` and `setLivesCount`, which would update DOM. Speaking about DOM updates: this is a heavy operation so in modern applications you want to make as little DOM manipulations as possible, so after score/life update you would probably re-render the score board in one go (not score and then lives separately).